scrap_reason_code/tests/test_scrap_reason_code.py

```diff
@@ -106,17 +106,17 @@ def test_scrap_reason_code(self):
         scrapped_move = picking.move_ids.filtered(lambda m: m.state == "done")
         self.assertTrue(scrapped_move, "No scrapped move created.")
         self.assertEqual(
-            scrapped_move.scrap_ids.ids, [scrap.id], "Wrong scrap linked to the move."
+            scrapped_move.scrap_id.id, scrap.id, "Wrong scrap linked to the move."
         )
         self.assertEqual(
             scrap.scrap_qty,
             5,
             "Scrap quantity has been modified and is not " "correct anymore.",
         )
-        move = scrap.move_id
-        self.assertEqual(move.reason_code_id.id, self.reason_code.id)
+        moves = scrap.move_ids
+        self.assertEqual(moves.mapped("reason_code_id.id"), [self.reason_code.id])
```

Comment:
FYI, Odoo supports mapping through all relational records with dot. So, `moves.reason_code_id.id` will work.

Reply:
@rousseldenis Thanks for looking into it.
I know that, but isn't there a technical possibility that moves are more than one record and moves.reason_code_id.id might end up in a singleton error?

Reply:
@john-herholz-dt The only case where you have that is if you put in the chain (without mapped()) another field (Boolean, Selection, Char,...)